### DISCUSSION TITLE
Combobox additions

### DISCUSSION THREAD

**Discussion by azim-b** [h2oai/wave#1135] [Ideas] [answered]
It'd be useful to have an option to put the `ui.combobox()`'s label and choices inline.

Also, could a `trigger` be set for a combobox? Use case example:

Present a combobox to select a `Warehouse`. Once user selects the warehouse, depending on the selection, show `Databases`. Based on  the database selection, show `Tables`, etc.

**Comment by mturoci** [comment, marked as answer]
> It'd be useful to have an option to put the ui.combobox()'s label and choices inline.

Not sure. If we did, we should do it for all other components + since our target audience are not regular UI developers, I am afraid to see apps where some inputs have labels inlined and some not, causing inconsistency, thus bad UX.

> Also, could a trigger be set for a combobox? Use case example:

Agree, tracked in https://github.com/h2oai/wave/issues/1136.

> Present a combobox to select a Warehouse. Once user selects the warehouse, depending on the selection, show Databases. Based on the database selection, show Tables, etc.

Combobox allows free-form text also meaning it will not work for cases when a specific value is needed (like in your case to show other inputs based on it). I suggest using dropdown instead.

**Reply by azim-b** [reply]
> ... causing inconsistency, thus bad UX.

gotcha!

> Agree, tracked in #1136.

thanks!

> Combobox allows free-form text also meaning it will not work for cases when a specific value is needed (like in your case to show other inputs based on it). I suggest using dropdown instead.

dropdown choices are getting too large. Plus the width of the dropdown is too wide. It's becoming harder to find the choice quickly. If trigger is set to a combobox, we could implement this with the combobox, correct?

**Reply by azim-b** [reply]
@mturoci example



https://user-images.githubusercontent.com/64787868/144131097-75c1b384-ad68-475c-9afb-09b54dc8bc64.mov

**Reply by mturoci** [reply]
>dropdown choices are getting too large. Plus the width of the dropdown is too wide. It's becoming harder to find the choice quickly. If trigger is set to a combobox, we could implement this with the combobox, correct?

I assume you have < 100 options in dropdown, which doesn't trigger a dialog with search. In that case, what you might be looking for can be https://github.com/h2oai/wave/issues/1116.
The combobox is not viable as you expect a one of the predefined values. With a combobox, users can type arbitrary text, e.g. `Warehouse1` which might not exist, thus crashing your app.

EDIT:

On second thought, #1116 will not help since you already have a dialog present.
